internal/auth/options.go

```diff
@@ -129,20 +131,82 @@ type SignatureData struct {
 	key  string
 }
 
-// NewOptions returns new options
-func NewOptions() *Options {
-	return &Options{
-		Port:            4180,
-		CookieName:      "_sso_auth",
-		CookieSecure:    true,
-		CookieHTTPOnly:  true,
-		CookieExpire:    time.Duration(168) * time.Hour,
-		CookieRefresh:   time.Duration(0),
-		SetXAuthRequest: false,
-		PassUserHeaders: true,
-		PassHostHeader:  true,
-		ApprovalPrompt:  "force",
-		RequestLogging:  true,
+// NewOptions returns new options with the below overrides
+func NewOptions() (*Options, error) {
+	overrides := map[string]interface{}{
+		"port":             4180,
+		"cookie_name":      "_sso_auth",
+		"cookie_secure":    true,
+		"cookie_http_only": true,
+		"cookie_expire":    "168h",
+		"cookie_refresh":   "1h",
+		"set_xauthrequest": false,
+		"passUser_headers": true,
+		"passHost_header":  true,
+		"approval_prompt":  "force",
+		"request_logging":  true,
+	}
+	options, err := loadVars(overrides)
+	if err != nil {
+		return nil, err
+	}
+	return options, nil
+}
+
+func loadVars(overrides map[string]interface{}) (*Options, error) {
+	var opts Options
+
+	bindAllOptVars(reflect.TypeOf(&opts).Elem(), "mapstructure")
+	setDefaults()
+
+	for key, value := range overrides {
+		viper.Set(key, value)
+	}
+
+	err := viper.Unmarshal(&opts)
+	if err != nil {
+		return nil, fmt.Errorf("unable to decode env vars into options struct")
+	}
+	return &opts, nil
+}
+
+// bindAllOptVars takes in a struct with tags and uses the tag values to bind env vars
+func bindAllOptVars(t reflect.Type, tag string) error {
```

Comment:
Not necessarily required with name-spacing (setting prefix will allow a similar result)

Reply:
This will be addressed, if required, in a future PR. For now - it makes sense with the included changes.